src/kem.rs

```diff
@@ -93,9 +111,21 @@ pub fn encapsulate(
     let (new_sk, new_pk) = key_gen(alg, rng)?;
     match alg {
         Algorithm::X25519 | Algorithm::Secp256r1 => {
-            let gxy = ecdh::derive(alg.into(), pk, &new_sk)?;
+            let gxy = ecdh::derive(alg.try_into().unwrap(), pk, &new_sk)?;
             Ok((gxy, new_pk))
         }
+
+        Algorithm::Kyber768 => {
+            let mut seed = [0; kyber768::SHARED_SECRET_SIZE];
+            rng.try_fill_bytes(&mut seed).map_err(|_| Error::KeyGen)?;
+
+            if let Ok((ct, ss)) = kyber768::encapsulate(pk.try_into().unwrap(), seed) {
```

Comment:
Lets not unwrap the key conversions. They really could panic if the API isn't used correctly.
Same in `decapsulate`.